In [77]:
import graphlab

In [78]:
products = graphlab.SFrame('amazon_baby.gl')

In [79]:
products['word_count'] = graphlab.text_analytics.count_words(products['review'])

In [80]:
products.head()

Columns:
	name	str
	review	str
	rating	float
	word_count	dict

Rows: 10

Data:
+-------------------------------+-------------------------------+--------+
|              name             |             review            | rating |
+-------------------------------+-------------------------------+--------+
|    Planetwise Flannel Wipes   | These flannel wipes are OK... |  3.0   |
|     Planetwise Wipe Pouch     | it came early and was not ... |  5.0   |
| Annas Dream Full Quilt wit... | Very soft and comfortable ... |  5.0   |
| Stop Pacifier Sucking with... | This is a product well wor... |  5.0   |
| Stop Pacifier Sucking with... | All of my kids have cried ... |  5.0   |
| Stop Pacifier Sucking with... | When the Binky Fairy came ... |  5.0   |
| A Tale of Baby's Days with... | Lovely book, it's bound ti... |  4.0   |
| Baby Tracker&reg; - Daily ... | Perfect for new parents. W... |  5.0   |
| Baby Tracker&reg; - Daily ... | A friend of mine pinned th... |  5.0   |
| Baby Tracker&reg; - Daily ... | This has been an easy way ... |  4.0   |
+-------------------------------+-------------------------------+--------+
+-------------------------------+
|           word_count          |
+-------------------------------+
| {'and': 5L, 'stink': 1L, '... |
| {'and': 3L, 'love': 1L, 'i... |
| {'and': 2L, 'quilt': 1L, '... |
| {'ingenious': 1L, 'and': 3... |
| {'and': 2L, 'parents!!': 1... |
| {'and': 2L, 'cute': 1L, 'h... |
| {'shop': 1L, 'be': 1L, 'is... |
| {'feeding,': 1L, 'and': 2L... |
| {'and': 1L, 'help': 1L, 'g... |
| {'journal.': 1L, 'all': 1L... |
+-------------------------------+
[10 rows x 4 columns]

In [81]:
# remove rating 3 review
products = products[products['rating'] != 3]

In [82]:
# positive sentiments rating >= 4 
products['sentiment'] = products['rating'] >= 4 

In [83]:
selected_words = ['awesome', 'great', 'fantastic', 'amazing', 'love', 'horrible', 'bad', 'terrible', 'awful', 'wow', 'hate']

In [84]:
def count_the_word0(inputRow):
    return inputRow['review'].count(selected_words[0])

def count_the_word1(inputRow):
    return inputRow['review'].count(selected_words[1])

def count_the_word2(inputRow):
    return inputRow['review'].count(selected_words[2])

def count_the_word3(inputRow):
    return inputRow['review'].count(selected_words[3])

def count_the_word4(inputRow):
    return inputRow['review'].count(selected_words[4])

def count_the_word5(inputRow):
    return inputRow['review'].count(selected_words[5])

def count_the_word6(inputRow):
    return inputRow['review'].count(selected_words[6])

def count_the_word7(inputRow):
    return inputRow['review'].count(selected_words[7])

def count_the_word8(inputRow):
    return inputRow['review'].count(selected_words[8])

def count_the_word9(inputRow):
    return inputRow['review'].count(selected_words[9])

def count_the_word10(inputRow):
    return inputRow['review'].count(selected_words[10])
        
        
        

In [85]:
products[selected_words[0]] = products.apply(count_the_word0)

products[selected_words[1]] = products.apply(count_the_word1)

products[selected_words[2]] = products.apply(count_the_word2)

products[selected_words[3]] = products.apply(count_the_word3)

products[selected_words[4]] = products.apply(count_the_word4)

products[selected_words[5]] = products.apply(count_the_word5)

products[selected_words[6]] = products.apply(count_the_word6)

products[selected_words[7]] = products.apply(count_the_word7)

products[selected_words[8]] = products.apply(count_the_word8)

products[selected_words[9]] = products.apply(count_the_word9)

products[selected_words[10]] = products.apply(count_the_word10)

In [86]:
products.head()

Columns:
	name	str
	review	str
	rating	float
	word_count	dict
	sentiment	int
	awesome	int
	great	int
	fantastic	int
	amazing	int
	love	int
	horrible	int
	bad	int
	terrible	int
	awful	int
	wow	int
	hate	int

Rows: 10

Data:
+-------------------------------+-------------------------------+--------+
|              name             |             review            | rating |
+-------------------------------+-------------------------------+--------+
|     Planetwise Wipe Pouch     | it came early and was not ... |  5.0   |
| Annas Dream Full Quilt wit... | Very soft and comfortable ... |  5.0   |
| Stop Pacifier Sucking with... | This is a product well wor... |  5.0   |
| Stop Pacifier Sucking with... | All of my kids have cried ... |  5.0   |
| Stop Pacifier Sucking with... | When the Binky Fairy came ... |  5.0   |
| A Tale of Baby's Days with... | Lovely book, it's bound ti... |  4.0   |
| Baby Tracker&reg; - Daily ... | Perfect for new parents. W... |  5.0   |
| Baby Tracker&reg; - Daily ... | A friend of mine pinned th... |  5.0   |
| Baby Tracker&reg; - Daily ... | This has been an easy way ... |  4.0   |
| Baby Tracker&reg; - Daily ... | I love this journal and ou... |  4.0   |
+-------------------------------+-------------------------------+--------+
+-------------------------------+-----------+---------+-------+-----------+---------+
|           word_count          | sentiment | awesome | great | fantastic | amazing |
+-------------------------------+-----------+---------+-------+-----------+---------+
| {'and': 3L, 'love': 1L, 'i... |     1     |    0    |   0   |     0     |    0    |
| {'and': 2L, 'quilt': 1L, '... |     1     |    0    |   0   |     0     |    0    |
| {'ingenious': 1L, 'and': 3... |     1     |    0    |   0   |     0     |    0    |
| {'and': 2L, 'parents!!': 1... |     1     |    0    |   1   |     0     |    0    |
| {'and': 2L, 'cute': 1L, 'h... |     1     |    0    |   1   |     0     |    0    |
| {'shop': 1L, 'be': 1L, 'is... |     1     |    0    |   0   |     0     |    0    |
| {'feeding,': 1L, 'and': 2L... |     1     |    0    |   0   |     0     |    0    |
| {'and': 1L, 'help': 1L, 'g... |     1     |    0    |   0   |     1     |    0    |
| {'journal.': 1L, 'all': 1L... |     1     |    0    |   0   |     0     |    0    |
| {'all': 1L, 'forget': 1L, ... |     1     |    0    |   0   |     0     |    0    |
+-------------------------------+-----------+---------+-------+-----------+---------+
+------+----------+-----+----------+-------+-----+------+
| love | horrible | bad | terrible | awful | wow | hate |
+------+----------+-----+----------+-------+-----+------+
|  1   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  3   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  2   |    0     |  0  |    0     |   0   |  0  |  0   |
+------+----------+-----+----------+-------+-----+------+
[10 rows x 16 columns]

In [87]:
products[selected_words[0]].sum()

3211L

In [88]:
products[selected_words[1]].sum()

47550L

In [89]:
products[selected_words[2]].sum()

1516L

In [90]:
products[selected_words[3]].sum()

2436L

In [91]:
products[selected_words[4]].sum()

62154L

In [92]:
products[selected_words[5]].sum()

958L

In [93]:
products[selected_words[6]].sum()

4320L

In [94]:
products[selected_words[7]].sum()

999L

In [95]:
products[selected_words[8]].sum()

627L

In [96]:
products[selected_words[9]].sum()

117L

In [97]:
products[selected_words[10]].sum()

3379L

# sentiment model

In [98]:
train_data,test_data = products.random_split(.8, seed=0)

In [99]:
selected_words_model = graphlab.logistic_classifier.create(train_data,
                                                    target = 'sentiment',
                                                     features = selected_words,
                                                     validation_set = test_data) 

Logistic regression:

--------------------------------------------------------

Number of examples          : 133448

Number of classes           : 2

Number of feature columns   : 11

Number of unpacked features : 11

Number of coefficients    : 12

Starting Newton Method

--------------------------------------------------------

+-----------+----------+--------------+-------------------+---------------------+

| Iteration | Passes   | Elapsed Time | Training-accuracy | Validation-accuracy |

+-----------+----------+--------------+-------------------+---------------------+

| 1         | 2        | 0.235175     | 0.845820          | 0.844193            |

| 2         | 3        | 0.391295     | 0.845775          | 0.844253            |

| 3         | 4        | 0.532399     | 0.845873          | 0.844313            |

| 4         | 5        | 0.686020     | 0.845865          | 0.844313            |

| 5         | 6        | 0.831124     | 0.845873          | 0.844313            |

| 6         | 7        | 0.974733     | 0.845873          | 0.844313            |

+-----------+----------+--------------+-------------------+---------------------+

SUCCESS: Optimal solution found.

In [100]:
selected_words_model['coefficients'].print_rows(num_rows=12)

+-------------+-------+-------+-----------------+------------------+
|     name    | index | class |      value      |      stderr      |
+-------------+-------+-------+-----------------+------------------+
| (intercept) |  None |   1   |  1.30429674772  | 0.00908064109618 |
|   awesome   |  None |   1   |  1.02212411097  | 0.0871664581374  |
|    great    |  None |   1   |  0.754258263477 | 0.0194083552903  |
|  fantastic  |  None |   1   |  0.793051167628 |  0.112391176958  |
|   amazing   |  None |   1   |  1.07317645858  |  0.100456419415  |
|     love    |  None |   1   |  1.20076740713  | 0.0210488901775  |
|   horrible  |  None |   1   |  -2.15736763241 | 0.0842502747482  |
|     bad     |  None |   1   |  -1.00686704748 | 0.0376576563717  |
|   terrible  |  None |   1   |  -2.07373008049 |  0.079396119601  |
|    awful    |  None |   1   |  -1.90936283887 |  0.100911763821  |
|     wow     |  None |   1   | -0.667484764099 |  0.258120224545  |
|     hate    |  None |   1   |  -

In [116]:
selected_words_model['coefficients'].sort('value',ascending=False).print_rows(num_rows=14)

+-------------+-------+-------+-----------------+------------------+
|     name    | index | class |      value      |      stderr      |
+-------------+-------+-------+-----------------+------------------+
| (intercept) |  None |   1   |  1.30429674772  | 0.00908064109618 |
|     love    |  None |   1   |  1.20076740713  | 0.0210488901775  |
|   amazing   |  None |   1   |  1.07317645858  |  0.100456419415  |
|   awesome   |  None |   1   |  1.02212411097  | 0.0871664581374  |
|  fantastic  |  None |   1   |  0.793051167628 |  0.112391176958  |
|    great    |  None |   1   |  0.754258263477 | 0.0194083552903  |
|     wow     |  None |   1   | -0.667484764099 |  0.258120224545  |
|     hate    |  None |   1   |  -0.71674753335 | 0.0455125193256  |
|     bad     |  None |   1   |  -1.00686704748 | 0.0376576563717  |
|    awful    |  None |   1   |  -1.90936283887 |  0.100911763821  |
|   terrible  |  None |   1   |  -2.07373008049 |  0.079396119601  |
|   horrible  |  None |   1   |  -

In [101]:
selected_words_model.evaluate(test_data)

{'accuracy': 0.8443129954359837,
 'auc': 0.6907033503513523,
 'confusion_matrix': Columns:
 	target_label	int
 	predicted_label	int
 	count	int
 
 Rows: 4
 
 Data:
 +--------------+-----------------+-------+
 | target_label | predicted_label | count |
 +--------------+-----------------+-------+
 |      1       |        0        |  120  |
 |      0       |        0        |  263  |
 |      0       |        1        |  5065 |
 |      1       |        1        | 27856 |
 +--------------+-----------------+-------+
 [4 rows x 3 columns],
 'f1_score': 0.9148562326551389,
 'log_loss': 0.4000842667631508,
 'precision': 0.8461468363658455,
 'recall': 0.9957106090935087,
 'roc_curve': Columns:
 	threshold	float
 	fpr	float
 	tpr	float
 	p	int
 	n	int
 
 Rows: 100001
 
 Data:
 +-----------+----------------+-----+-------+------+
 | threshold |      fpr       | tpr |   p   |  n   |
 +-----------+----------------+-----+-------+------+
 |    0.0    |      1.0       | 1.0 | 27976 | 5328 |
 |   1e-05  

In [102]:
sentiment_model = graphlab.logistic_classifier.create(train_data,
                                                    target = 'sentiment',
                                                     features = ['word_count'],
                                                     validation_set = test_data) 

Logistic regression:

--------------------------------------------------------

Number of examples          : 133448

Number of classes           : 2

Number of feature columns   : 1

Number of unpacked features : 219217

Number of coefficients    : 219218

Starting L-BFGS

--------------------------------------------------------

+-----------+----------+-----------+--------------+-------------------+---------------------+

| Iteration | Passes   | Step size | Elapsed Time | Training-accuracy | Validation-accuracy |

+-----------+----------+-----------+--------------+-------------------+---------------------+

| 1         | 5        | 0.000002  | 1.346136     | 0.841481          | 0.839989            |

| 2         | 9        | 3.000000  | 3.001839     | 0.947425          | 0.894877            |

| 3         | 10       | 3.000000  | 3.504208     | 0.923768          | 0.866232            |

| 4         | 11       | 3.000000  | 4.002587     | 0.971779          | 0.912743            |

| 5         | 12       | 3.000000  | 4.521976     | 0.975511          | 0.908900            |

| 6         | 13       | 3.000000  | 5.191969     | 0.899991          | 0.825967            |

| 10        | 18       | 1.000000  | 7.608799     | 0.988715          | 0.916256            |

+-----------+----------+-----------+--------------+-------------------+---------------------+

TERMINATED: Iteration limit reached.

This model may not be optimal. To improve it, consider increasing `max_iterations`.

In [103]:
sentiment_model.evaluate(test_data)

{'accuracy': 0.916256305548883,
 'auc': 0.9446492867438502,
 'confusion_matrix': Columns:
 	target_label	int
 	predicted_label	int
 	count	int
 
 Rows: 4
 
 Data:
 +--------------+-----------------+-------+
 | target_label | predicted_label | count |
 +--------------+-----------------+-------+
 |      1       |        0        |  1461 |
 |      0       |        1        |  1328 |
 |      0       |        0        |  4000 |
 |      1       |        1        | 26515 |
 +--------------+-----------------+-------+
 [4 rows x 3 columns],
 'f1_score': 0.9500349343413533,
 'log_loss': 0.26106698432421777,
 'precision': 0.9523039902309378,
 'recall': 0.9477766657134686,
 'roc_curve': Columns:
 	threshold	float
 	fpr	float
 	tpr	float
 	p	int
 	n	int
 
 Rows: 100001
 
 Data:
 +-----------+----------------+----------------+-------+------+
 | threshold |      fpr       |      tpr       |   p   |  n   |
 +-----------+----------------+----------------+-------+------+
 |    0.0    |      1.0       | 

In [104]:
test_data.head()

Columns:
	name	str
	review	str
	rating	float
	word_count	dict
	sentiment	int
	awesome	int
	great	int
	fantastic	int
	amazing	int
	love	int
	horrible	int
	bad	int
	terrible	int
	awful	int
	wow	int
	hate	int

Rows: 10

Data:
+-------------------------------+-------------------------------+--------+
|              name             |             review            | rating |
+-------------------------------+-------------------------------+--------+
| Nature's Lullabies First Y... | Space for monthly photos, ... |  5.0   |
| Nature's Lullabies Second ... | This was the only calender... |  5.0   |
|  Lamaze Peekaboo, I Love You  | I like how the book has a ... |  5.0   |
|  Lamaze Peekaboo, I Love You  | What a great book for babi... |  5.0   |
| SoftPlay Twinkle Twinkle E... |                               |  5.0   |
| SoftPlay Peek-A-Boo Where'... | We wanted to get another b... |  4.0   |
| SoftPlay Peek-A-Boo Where'... | My 3 month old son loves t... |  5.0   |
|   Our Baby Girl Memory Book   | Beautiful book, I love it ... |  5.0   |
| God Created Everything Nit... | This is well made a very p... |  5.0   |
| Cloth Diaper Pins Stainles... | A great safety pin that ha... |  5.0   |
+-------------------------------+-------------------------------+--------+
+-------------------------------+-----------+---------+-------+-----------+---------+
|           word_count          | sentiment | awesome | great | fantastic | amazing |
+-------------------------------+-----------+---------+-------+-----------+---------+
| {'it.': 1L, 'info': 1L, 'w... |     1     |    0    |   0   |     0     |    0    |
| {'and': 1L, 'this': 2L, 't... |     1     |    0    |   0   |     0     |    0    |
| {'family': 1L, 'over': 1L,... |     1     |    0    |   0   |     0     |    0    |
| {'page.': 1L, "couldn't": ... |     1     |    0    |   1   |     0     |    0    |
|               {}              |     1     |    0    |   0   |     0     |    0    |
| {'and': 2L, 'another': 1L,... |     1     |    0    |   0   |     0     |    0    |
| {'and': 7L, 'old': 1L, 'el... |     1     |    0    |   0   |     0     |    0    |
| {'pink': 1L, 'pastel': 1L,... |     1     |    0    |   1   |     0     |    0    |
| {'and': 1L, 'is': 1L, 'gif... |     1     |    0    |   1   |     0     |    0    |
| {'and': 1L, 'a': 3L, 'grea... |     1     |    0    |   2   |     0     |    0    |
+-------------------------------+-----------+---------+-------+-----------+---------+
+------+----------+-----+----------+-------+-----+------+
| love | horrible | bad | terrible | awful | wow | hate |
+------+----------+-----+----------+-------+-----+------+
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  1   |    0     |  0  |    0     |   0   |  0  |  0   |
|  1   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |
+------+----------+-----+----------+-------+-----+------+
[10 rows x 16 columns]

In [105]:
test_data_positive = test_data[test_data['sentiment'] == 1]

In [106]:
test_data_positive.num_rows()

27976

In [107]:
test_data.num_rows()

33304

In [108]:
27976.0/33304.0

0.8400192169108815

In [121]:
diaper_champ_reviews = products[products['name'] == "Baby Trend Diaper Champ"]

In [122]:
diaper_champ_reviews['predicted_sentiment'] = sentiment_model.predict(diaper_champ_reviews,output_type='probability')

In [123]:
diaper_champ_reviews= diaper_champ_reviews.sort('predicted_sentiment',ascending=False)

In [124]:
diaper_champ_reviews.head()

Columns:
	name	str
	review	str
	rating	float
	word_count	dict
	sentiment	int
	awesome	int
	great	int
	fantastic	int
	amazing	int
	love	int
	horrible	int
	bad	int
	terrible	int
	awful	int
	wow	int
	hate	int
	predicted_sentiment	float

Rows: 10

Data:
+-------------------------+-------------------------------+--------+
|           name          |             review            | rating |
+-------------------------+-------------------------------+--------+
| Baby Trend Diaper Champ | Baby Luke can turn a clean... |  5.0   |
| Baby Trend Diaper Champ | I LOOOVE this diaper pail!... |  5.0   |
| Baby Trend Diaper Champ | We researched all of the d... |  4.0   |
| Baby Trend Diaper Champ | My baby is now 8 months an... |  5.0   |
| Baby Trend Diaper Champ | This is absolutely, by far... |  5.0   |
| Baby Trend Diaper Champ | Diaper Champ or Diaper Gen... |  5.0   |
| Baby Trend Diaper Champ | Wow!  This is fabulous.  I... |  5.0   |
| Baby Trend Diaper Champ | I originally put this item... |  5.0   |
| Baby Trend Diaper Champ | Two girlfriends and two fa... |  5.0   |
| Baby Trend Diaper Champ | I am one of those super-cr... |  5.0   |
+-------------------------+-------------------------------+--------+
+-------------------------------+-----------+---------+-------+-----------+---------+
|           word_count          | sentiment | awesome | great | fantastic | amazing |
+-------------------------------+-----------+---------+-------+-----------+---------+
| {'all': 1L, 'less': 1L, "f... |     1     |    0    |   0   |     0     |    0    |
| {'just': 1L, 'over': 1L, '... |     1     |    0    |   0   |     0     |    0    |
| {'all': 2L, 'just': 4L, "d... |     1     |    0    |   0   |     0     |    0    |
| {"don't": 1L, 'when': 1L, ... |     1     |    0    |   2   |     0     |    0    |
| {'just': 3L, 'money': 1L, ... |     1     |    0    |   0   |     0     |    0    |
| {'all': 1L, 'bags.': 1L, '... |     1     |    0    |   0   |     0     |    0    |
| {'and': 4L, '"genie".': 1L... |     1     |    0    |   0   |     0     |    0    |
| {'lysol': 1L, 'all': 2L, '... |     1     |    0    |   0   |     0     |    0    |
| {'just': 1L, 'when': 1L, '... |     1     |    0    |   0   |     0     |    1    |
| {'taller': 1L, 'bags.': 1L... |     1     |    0    |   0   |     0     |    0    |
+-------------------------------+-----------+---------+-------+-----------+---------+
+------+----------+-----+----------+-------+-----+------+---------------------+
| love | horrible | bad | terrible | awful | wow | hate | predicted_sentiment |
+------+----------+-----+----------+-------+-----+------+---------------------+
|  1   |    0     |  0  |    0     |   0   |  0  |  0   |    0.999999937267   |
|  1   |    0     |  0  |    0     |   0   |  0  |  0   |    0.999999917406   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |    0.999999899509   |
|  0   |    0     |  1  |    0     |   0   |  0  |  0   |    0.999999836182   |
|  2   |    0     |  0  |    0     |   0   |  0  |  0   |    0.999999824745   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |    0.999999759315   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |    0.999999692111   |
|  0   |    0     |  0  |    0     |   0   |  0  |  0   |    0.999999642488   |
|  0   |    1     |  0  |    0     |   0   |  0  |  0   |    0.999999604504   |
|  1   |    0     |  0  |    0     |   0   |  0  |  0   |    0.999999486804   |
+------+----------+-----+----------+-------+-----+------+---------------------+
[10 rows x 17 columns]

In [125]:
selected_words_model.predict(diaper_champ_reviews[0:1], output_type='probability')

dtype: float
Rows: 1
[0.9244960743879546]

In [126]:
diaper_champ_reviews[0:1]

Columns:
	name	str
	review	str
	rating	float
	word_count	dict
	sentiment	int
	awesome	int
	great	int
	fantastic	int
	amazing	int
	love	int
	horrible	int
	bad	int
	terrible	int
	awful	int
	wow	int
	hate	int
	predicted_sentiment	float

Rows: 1

Data:
+-------------------------+-------------------------------+--------+
|           name          |             review            | rating |
+-------------------------+-------------------------------+--------+
| Baby Trend Diaper Champ | Baby Luke can turn a clean... |  5.0   |
+-------------------------+-------------------------------+--------+
+-------------------------------+-----------+---------+-------+-----------+---------+
|           word_count          | sentiment | awesome | great | fantastic | amazing |
+-------------------------------+-----------+---------+-------+-----------+---------+
| {'all': 1L, 'less': 1L, "f... |     1     |    0    |   0   |     0     |    0    |
+-------------------------------+-----------+---------+-------+-----------+---------+
+------+----------+-----+----------+-------+-----+------+---------------------+
| love | horrible | bad | terrible | awful | wow | hate | predicted_sentiment |
+------+----------+-----+----------+-------+-----+------+---------------------+
|  1   |    0     |  0  |    0     |   0   |  0  |  0   |    0.999999937267   |
+------+----------+-----+----------+-------+-----+------+---------------------+
[1 rows x 17 columns]

# First attempt

# Out of the 11 words in selected_words, which one is most used in the reviews in the dataset? love (X) -> great

# Out of the 11 words in selected_words, which one is least used in the reviews in the dataset ? wow

# Out of the 11 words in selected_words, which one got the most positive weight in the selected_words_model? love

# Out of the 11 words in selected_words, which one got the most negative weight in the selected_words_model? horrible (X) -> terrible

# Which of the following ranges contains the accuracy of the selected_words_model on the test_data? 0.841 - 0.871

# Which of the following ranges contains the accuracy of the sentiment_model in the IPython Notebook from lecture on the test_data? 0.901 - 0.931

# Which of the following ranges contains the accuracy of the majority class classifier, which simply predicts the majority class on the test_data? 0.811-0.843

# How do you compare the different learned models with the baseline approach where we are just predicting the majority class?all word > the two. other two same

# Which of the following ranges contains the ‘predicted_sentiment’ for the most positive review for ‘Baby Trend Diaper Champ’, according to the sentiment_model from the IPython Notebook from lecture? 0.9 - 1.0

# Consider the most positive review for ‘Baby Trend Diaper Champ’ according to the sentiment_model from the IPython Notebook from lecture. Which of the following ranges contains the predicted_sentiment for this review, if we use the selected_words_model to analyze it? 0.7 0.8

# Why is the value of the predicted_sentiment for the most positive review found using the sentiment_model much more positive than the value predicted using the selected_words_model? review is positive, too many negative word in selected_word  (X) -> none of